extensions/azure/storage/AzureDataLakeStorage.cpp

```diff
@@ -72,4 +77,62 @@ std::optional<uint64_t> AzureDataLakeStorage::fetchFile(const FetchAzureDataLake
   }
 }
 
+bool AzureDataLakeStorage::matchesPathFilter(const std::string& base_directory, const std::string& path_filter, std::string path) {
+  if (path_filter.empty()) {
+    return true;
+  }
+
+  if (!base_directory.empty()) {
+    gsl_Expects(minifi::utils::StringUtils::startsWith(path, base_directory));
```

Comment:
A precondition should preferably be declared at the start of the function body with `gsl_Expects`.
You may want an implication (base_directory is not empty -> path starts with base_directory), which doesn't have a logical operator in C++, so I think it would be best to add a function `bool implies(bool a, bool b) noexcept { return !a || b; }` (maybe in GeneralUtils.h?) and use it in the declaration.

Reply:
Updated in ef0172d119c472a1c91a10aecbe531a112ade2bf